prow/cmd/deck/job_history.go

```diff
@@ -240,16 +256,39 @@ func (bucket gcsBucket) listBuildIDs(root string) ([]int64, error) {
 	return ids, nil
 }
 
-func parseJobHistURL(url *url.URL) (bucketName, root string, buildID int64, err error) {
+// parseJobHistURL parses the job History URL
+// example urls:
+// * new format: https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-capi?buildId=1245584383100850177
+// * old format: https://prow.k8s.io/job-history/kubernetes-jenkins/pr-logs/directory/pull-capi?buildId=1245584383100850177
+// Newly generated URLs will include the storageProvider. We still support old URLs so they don't break.
+// For old URLs we assume that the storageProvider is `gs`.
+// examples return values:
+// * storageProvider: s3, gs
```

Comment:
Wouldn't it make sense to sort these alphabetically?

Reply:
Done